x-pack/plugins/index_management/public/hooks/use_details_page_mappings_model_management.ts

```diff
@@ -0,0 +1,139 @@
+  const {
+    plugins: { ml },
+  } = useAppContext();
+
+  const { api } = useComponentTemplatesContext();
+
+  const { inferenceToModelIdMap } = state;
+
+  const [pendingDeployments, setPendingDeployments] = useState<Set<string>>(new Set());
+
+  const fetchInferenceModelsAndTrainedModelStats = useCallback(async () => {
+    const inferenceModels = (await api.getInferenceModels()) as InferenceModel;
+    const trainedModelStats = await ml?.mlApi?.trainedModels.getTrainedModelStats();
+
+    return { inferenceModels, trainedModelStats };
+  }, [api, ml]);
+
+  const getTrainedModelStats = useCallback(
+    (modelStats?: InferenceStatsResponse): DeploymentStatusType => {
+      const deploymentStatsByModelId =
+        modelStats?.trained_model_stats.reduce<DeploymentStatusType>((acc, modelStat) => {
+          if (modelStat.model_id) {
+            acc[modelStat.model_id] =
+              modelStat?.deployment_stats?.state === 'started' ? 'deployed' : 'not_deployed';
+          }
+          return acc;
+        }, {}) ?? {};
+
+      return deploymentStatsByModelId;
+    },
+    []
+  );
+
+  const getDefaultInferenceIds = useCallback((deploymentStatsByModelId: DeploymentStatusType) => {
+    return {
+      elser_model_2: {
+        trainedModelId: '.elser_model_2',
+        isDeployed: deploymentStatsByModelId['.elser_model_2'] === 'deployed',
+        defaultInferenceEndpoint: true,
+      },
+      e5: {
+        trainedModelId: '.multilingual-e5-small',
+        isDeployed: deploymentStatsByModelId['.multilingual-e5-small'] === 'deployed',
+        defaultInferenceEndpoint: true,
+      },
+    };
+  }, []);
+
+  const getCustomInferenceIdMap = useCallback(
+    (deploymentStatsByModelId: DeploymentStatusType, models?: InferenceModel) => {
+      return models?.data.reduce<InferenceToModelIdMap>((inferenceMap, model) => {
+        const inferenceId = model.model_id;
+        const trainedModelId =
+          'model_id' in model.service_settings ? model.service_settings.model_id : '';
+        inferenceMap[inferenceId] = {
+          trainedModelId,
+          isDeployed: deploymentStatsByModelId[trainedModelId] === 'deployed',
+          defaultInferenceEndpoint: false,
+        };
+        return inferenceMap;
+      }, {});
+    },
+    []
+  );
+
+  const getInferenceToModelIdMap = useCallback(async () => {
+    const { inferenceModels, trainedModelStats } = await fetchInferenceModelsAndTrainedModelStats();
+
+    const deploymentStatsByModelId = getTrainedModelStats(trainedModelStats);
+    const defaultInferenceIds = getDefaultInferenceIds(deploymentStatsByModelId);
+    const modelIdMap = getCustomInferenceIdMap(deploymentStatsByModelId, inferenceModels);
+    return { ...defaultInferenceIds, ...modelIdMap };
+  }, [
+    getCustomInferenceIdMap,
+    getDefaultInferenceIds,
+    getTrainedModelStats,
+    fetchInferenceModelsAndTrainedModelStats,
+  ]);
+
+  const getInferenceIdsInPendingList = useCallback(() => {
```

Comment:
This can just be a `const` assignment using useMemo, instead of a `get` function that needs to be actively called. React will keep it up to date. And: use Arrays instead of sets. It'll make your life easier throughout this code.

Reply:
Updated. Using array instead of set